### PR TITLE
Use an SDK client to start workflows

### PR DESCRIPTION
**Pull request by robholland**
**What changed?**
tctl now uses an SDK client rather than a service client to start workflows.

**Why?**

Using the SDK client brings the tctl code more inline with standard way
for users to start workflows and makes it easier for contributors to
adjust tctl and make use of SDK features such as interceptors/context
propagation.

This will also be useful in future if we wish to provide more plugin
APIs to control the SDK client.

This also avoids using more complex code paths for parsing memo and
search attributes given that we can only be receiving untyped JSON
inputs from the command line anyway.

**How did you test it?**
Tests were updated. Also manually tested error handling for invalid memo/search inputs.

**Potential risks**

**Is hotfix candidate?**

No.

### CODE REVIEW

**Review by yiminc — APPROVED** [review]

File: tools/cli/workflowCommands.go [hunk 1]
```diff
@@ -297,29 +313,33 @@ func processSearchAttr(c *cli.Context) *commonpb.SearchAttributes {
...
+			ErrorAndExit("Search attribute JSON parse error.", err)
+		}
+		searchAttrValues = append(searchAttrValues, j)
 	}
 
-	return searchAttributes
+	fields := make(map[string]interface{}, len(searchAttrKeys))
+	for i, key := range searchAttrKeys {
+		fields[key] = searchAttrValues[i]
+	}
```
Comment (alexshtin) [inline tools/cli/workflowCommands.go:338]: nit: I think you can combine these two `for` loops into the one.

File: tools/cli/workflowCommands.go [hunk 2]
```diff
@@ -297,29 +313,33 @@ func processSearchAttr(c *cli.Context) *commonpb.SearchAttributes {
...
+		ErrorAndExit(fmt.Sprintf("Uneven number of search attributes keys (%d): %v and values(%d): %v.", len(searchAttrKeys), searchAttrKeys, len(rawSearchAttrVals), rawSearchAttrVals), nil)
 	}
 
-	searchAttributesStr := make(map[string]string, len(searchAttrKeys))
-	for i, searchAttrVal := range searchAttrVals {
-		searchAttributesStr[searchAttrKeys[i]] = searchAttrVal
-	}
+	var searchAttrValues []interface{}
 
-	searchAttributes, err := searchattribute.Parse(searchAttributesStr, nil)
```
Comment (alexshtin) [inline tools/cli/workflowCommands.go:314]: So you don't want to validate search attributes on the CLI side anymore. You just unmarshal them using json (this is a contract between CLI user and CLI) and send them to server. Server will do all names and types validation and return an error if it fails. I think this is a good approach.

**Review by alexshtin — APPROVED** [review]
1. Can you test it (at least manually) with different search attribute types: `keyword`, `double`, `datetime`, and `int`?
2. What's about other frontend calls (signal, query, describe, etc)? Shouldn't they also be replaced with SDK and frontend client completely removed from tctl?

**Comment by robholland** [comment]
Yes I'd like to replace all possible use cases with SDK, just didn't want to do all that in one PR and/or before running this PR past the team to check we're happy with the approach.

**Comment by robholland** [comment]
Manually tested that the various types @alexshtin mentioned accept valid values and show errors for invalid formats, string vs number, valid date time format vs junk, etc.

**Pull request merged**